sstables/mp_row_consumer.cc

```diff
@@ -101,6 +128,10 @@ std::vector<column_translation::column_info> column_translation::state::build(
         for (auto&& desc : src) {
             const bytes& type_name = desc.type_name.value;
             data_type type = db::marshal::type_parser::parse(to_sstring_view(type_name));
+            if (!features.is_enabled(CorrectUDTsInCollections)) {
+                // See #6130.
+                type = freeze_types_in_collections(std::move(type));
+            }
```

Comment:
So here you should check that scylla metadata is present, otherwise it could be a Cassandra sstable. A lazy way to do it is to check whether any bit is enabled in features, hoping there is no sstable so old that it has no bits set (a safe bet since mc format came after feature bits) and that we don't start retiring old feature bits (less safe).

A less lazy way is to add a separate boolean to sstable_enabled_features indicating that the metadata itself was present.

Reply:
Done. I also updated the PR cover letter to explain what happens.